File: src/java/KrishnamurthyNumber.java
package java;

public class KrishnamurthyNumber {

    private static long factorial(long num) {
        long fact = 1;
        while(num > 0)
            fact *= num--;
        return fact;
    }

    public static boolean isKrishnamurthyNumber(long num) {
        long tempNum = num;
        long sum = 0;

        while(tempNum > 0) {
            sum += factorial(tempNum % 10);
            tempNum /= 10;
        }

        return sum == num;
    }

}

File: src/C++/KrishnamurthyNumber.cpp
#include <iostream>
using namespace std;


int fact(long n) {
    int fact = 1;
    while (n > 0) {
        fact *= n--;
    }
    return fact;
}

bool isKrishanMurthyNumber(long n) {
    long temp = n;
    long sum = 0;
    while (temp > 0) {
        sum += fact(temp % 10);
        temp /= 10;
    }
    return sum == n;
}